### PR TITLE
Fixes #4778 - Enforcing SNI when there are only non-wildcards certifi…

### LINKED ISSUE #4778: Enforcing SNI when there are only non-wildcards certificates
Given the keystore file:

```keytool -keystore keystore -genkey -keyalg RSA -alias jetty```

and enforcing SNI:

```sslContextFactory.setSniRequired(true);```

I get the following error:

```java.lang.IllegalStateException: No SNI Key managers when SNI is required```

Adding second (non-wildcard) host doesn't change anything. However, it works fine with a wildcard certificate.
Is there something wrong with how I'm initializing keystore?

### PR DESCRIPTION
**Pull request by sbordet**
…cates.

If SNI is required, wrap the KeyManagers with SniX509ExtendedKeyManager.
Updated the main keystore file to only have one certificate (instead of two),
since there never was the need for two certificates in the tests.

Signed-off-by: Simone Bordet <simone.bordet@gmail.com>

Closes #4778.

### CODE REVIEW

File: jetty-util/src/main/java/org/eclipse/jetty/util/ssl/SslContextFactory.java [hunk 1]
```diff
@@ -1247,14 +1247,13 @@ protected KeyStore loadTrustStore(Resource resource) throws Exception
                 }
 
                 // Is SNI needed to select a certificate?
-                if (!_certWilds.isEmpty() || _certHosts.size() > 1 || (_certHosts.size() == 1 && _aliasX509.size() > 1))
+                boolean sniRequired = (this instanceof Server) && ((Server)this).isSniRequired();
+                if (sniRequired || !_certWilds.isEmpty() || _certHosts.size() > 1 || (_certHosts.size() == 1 && _aliasX509.size() > 1))
```
Comment (rnd4222) [inline jetty-util/src/main/java/org/eclipse/jetty/util/ssl/SslContextFactory.java:1251]: Maybe a stupid question, but is it possible to move `newSniX509ExtendedKeyManager` wrapping to a `Server.getKeyManagers` method?

Comment (sbordet) [reply jetty-util/src/main/java/org/eclipse/jetty/util/ssl/SslContextFactory.java:1251]: @rnd4222 we have done this in Jetty 10, but in 9.4.x there are people still using `SslContextFactory` (not `SslContextFactory.Server`).

**Comment by sbordet** [comment]
@gregw bump

**Comment by joakime** [comment]
Will this change impact the ability of `SecureRequestCustomizer.setSniRequired(boolean)` to function?

https://github.com/eclipse/jetty.project/blob/jetty-9.4.28.v20200408/jetty-server/src/main/java/org/eclipse/jetty/server/SecureRequestCustomizer.java#L133-L140

**Review by gregw — APPROVED** [review]

**Comment by sbordet** [comment]
@joakime I don't understand?

**Pull request merged**